src/server/terrain/structures/SimpleTreeModel.zig

```diff
@@ -83,13 +108,14 @@ pub fn generate(self: *SimpleTreeModel, x: i32, y: i32, z: i32, chunk: *main.chu
 					var py = chunk.startIndex(y + 1 - j);
 					while(py < y + j) : (py += chunk.super.pos.voxelSize) {
 						if(chunk.liesInChunk(px, py, pz))
-							chunk.updateBlockIfDegradable(px, py, pz, .{.typ = self.leavesBlock, .data = 0}); // TODO: Natural standard.
+							if (chunk.getBlock(px, py, pz).typ == 0)
```

Comment:
I think this could break the fog forest. We want to be able to place leaves when the air is filled with fog.
What are you trying to fix here?

Reply:
The leaves would replace the logs that are branched. I changed it so that it checks if it is transparent instead

Reply:
The branches wouldn't get replaced because they are not degradable and you are calling `updateBlockIfDegradable`. Otherwise the leaves would also replace the stem. Your check is unnecessary.